Tutorials/Level2Tutorial.md

```diff
@@ -0,0 +1,1081 @@
+# Level 2 Tutorial
+
+## Table of Contents
+1. [Parallel SST Runs](#ParallelSSTRuns)
+2. [Advanced Python Scripting](#AdvPythonScripting)
+3. [Advanced JSON Scripting](#AdvJSONScripting)
+4. [Loading Parallel Graphs](#LoadingParallelGraphs)
+5. [Links to External SST Components](#ExternSSTComp)
+
+## Parallel SST runs: <a name="ParallelSSTRuns"></a>
+
+### Introduction and Environment
+
+One of the key features of SST is the ability to execute parallel 
+simulations that utilize the MPI (Message Passing Interface) to span 
+multiple cores and/or multiple nodes in a cluster.  The SST Core interfaces 
+are natively built to utilize MPI in order to perform communication between 
+individual components across multiple MPI "ranks."  This enables users to 
+build and execute large scale simulations without fear of exhausting a single 
+system's resources.  When a user requests a parallel simulation (using MPI 
+or threading), the SST Core infrastructure will automatically partition the 
+requested components and subcomponents across the target number of parallel 
+units.  This partitioning considers the requested simulation components, the 
+connectivity of the components and the latency of the links between components 
+in order to co-locate components and subcomponents that are expected to frequently 
+exchange events.
```

Comment:
This is a bit of an overstatement. Maybe change to "in order to minimize synchronization overhead".

Some background: SST has a few built-in automatic partitioning options (run `sst-info sst` to view), in addition to accepting a manual partitioning. None of the automatic options take into account how frequently events are expected to be exchanged. That would be a good metric, as would co-locating components that interact more frequently, but the core doesn't have enough info to predict this. The `simple` partitioner does take into account the graph structure as it tries to cut high-latency links and spread components evenly across ranks/threads. The goal is to minimize synchronization frequency (cutting high latency links) while balancing work across ranks (number of components is a _very_ loose proxy for amount of work, but one that is easy to compute ahead of time).

Reply:
Duly noted.  This is an area where we are very interested to help in the future for very large graphs.  I made the change in the documentation though.